### ISSUE TITLE
MySQL Errors 1406 and 1366

### ISSUE THREAD

**Issue by Robinson-D** [gophish/gophish#1312]
First of all - awesome project. can see how much time and work has gone into it and i;m sure the whole community appreciates you providing it open source. 

What version of Gophish are you using?: 0.6 (i believe this is still applicable to 0.7 though)

Brief description of the issue: MySQL errors when saving HTML landing pages that are either too long > 16KB or include UTF-8 characters.

What are you expecting to see happen? : HTML landing page should save without any issues

What are you seeing happen? Landing page does not save and the following errors are noted: 

If the html content is too large, mysql throws a 1406  error code: 

     Error Code: 1406. Data too long for column

If the html content contains UTF-8 characters, mysql throws a 1366 error code: 

     Error Code: 1366. Incorrect string value: '\xF0\x9F\x94\x8D' for column

If this question is related to email templates or landing pages not working as expected, please provide your template or landing page below:

```html
For error 1406: Put in any large template that goes over the 16KB threshold for a text field. 
For error 1366: Put in any template with UTF-8 characters that take 4 bytes to be represented. 
```

I have been able to fix this issue by running the following command on the gophish mysql table: 

mysql> alter table pages modify html mediumtext character set utf8mb4;

This changes the current html field type from text to mediumtext, allowing the value to be up to 16MB instead of 16KB. 
Furthermore, it sets the character set to utf8mb4 which allows utf-8 characters to be represented by 4 bytes per code point. 

references: 
https://stackoverflow.com/questions/6766781/maximum-length-for-mysql-type-text
https://stackoverflow.com/questions/30074492/what-is-the-difference-between-utf8mb4-and-utf8-charsets-in-mysql
 
 
Going forward - when gophish generates the tables and columns for the mysql database, it would be awesome if it generated all html columns with the characteristics above. Uncertain how to incorporate this into your source code but hopefully the above might give another community member or yourself a starting point.

**Comment by jordan-wright** [comment]
Hey @Robinson-D,

Thanks for reaching out! You're absolutely right on both fronts, and the good news is that we've seen this before.

For the UTF-8 support in MySQL issue, you can find the previous discussion [here](https://github.com/gophish/gophish/issues/1249#issuecomment-433550237). I've updated our documentation to reflect the changes needed when creating the database. Let me know if you see something missing there.

I also have a [database migration](https://github.com/gophish/gophish/blob/master/db/db_mysql/migrations/20180609164054_0.7.0_update_html_storage.sql) that's active as of 0.7.0 which converts the HTML fields to `mediumtext` to prevent the issue you described.

So, in short, both of these issues are known and have been addressed, but thank you so much for reaching out with the detailed run down. Had we not seen these before, this kind of information would have been so incredibly helpful to track down the root cause.

Keep up the great work!
